src/lib/eventStack/EventSet.js

```diff
@@ -0,0 +1,65 @@
+export default class EventSet {
+  /**
+   * @param {Function[]} eventHandlers
+   */
+  constructor(eventHandlers) {
+    /** @private {Set<Function>} handlers */
+    this.handlers = new Set(eventHandlers)
+  }
+
+  /**
+   * @param {Function[]} eventHandlers
+   * @return {EventSet}
+   */
+  addHandlers(eventHandlers) {
+    const handlerSet = new Set(this.handlers)
+
+    eventHandlers.forEach((eventHandler) => {
+      // Heads up!
+      // We should delete a handler from the set, otherwise it will be not the last element in the
+      // set.
+      handlerSet.delete(eventHandler)
+      handlerSet.add(eventHandler)
+    })
+
+    return new EventSet(handlerSet)
+  }
+
+  /**
+   * @param {Event} event
+   * @param {Boolean} dispatchAll
+   */
+  dispatchEvent(event, dispatchAll) {
+    if (dispatchAll) {
+      this.handlers.forEach((handler) => {
+        handler(event)
+      })
+      return
+    }
+
+    const recentHandler = [...this.handlers].pop()
```

Comment:
Rather than cloning the entire array to get the last handler, you can directly access the last item:

```jsx
this.handlers.slice(-1)
```

Note, both the spread approach and the slice approach avoid mutating the `this.handlers` array, however, the `recentHandler` is not a unique instance.  It is the same instance as was originally provided to the array of handlers.  I'm not sure if this matters to you or not.